website/config/http.js

```diff
@@ -49,11 +49,18 @@ module.exports.http = {
     *                                                                          *
     ***************************************************************************/
 
-    // bodyParser: (function _configureBodyParser(){
-    //   var skipper = require('skipper');
-    //   var middlewareFn = skipper({ strict: true });
-    //   return middlewareFn;
-    // })(),
+    bodyParser: (function _configureBodyParser(){
+      var skipper = require('skipper');
+      var middlewareFn = skipper({
+        strict: true,
+        limit: '1mb',// [?] https://github.com/expressjs/body-parser/tree/ee91374eae1555af679550b1d2fb5697d9924109#limit-1
+        onBodyParserError: (err, req, res)=>{
+          // If an error occurs while parsing an incoming request body, return the statusCode and error message from the body parser
+          return res.status(err.status).send(err.message);
```

Comment:
If we're going to negotiate the error and respond with a bad request, then it's important to send a message back. Ideally it's a parsable message too, if the user agent wants JSON.  For example, `res.badRequest()`.

I think you also want to handle the case where it's not a recognized error, and then you want to use `res.serverError()` because that will obfuscate the error details from the user agent and use the default logging that alerts are tuned to, so that we find out.

This also means it's noisy and we hear about it if an unrecognized/unexpected error happens.

```js
if (_.isNumber(err.statusCode) && err.statusCode >= 400 && err.statusCode <500) {
  return res.badRequest(err);
} else {
  return res.serverError(err);
}
```

Reply:
```
sails.log.error(err);// but include whatever preamble is required for alerting (see default implementation of serverError())
res.status(500).send();
```

for the badrequest one you can probably just do it way you were

```
res.status(400).send(err.message);
```